Fail intercepted HTLC backwards when we should

luckysori opened the pull request:
Fixes #943.
Fixes #944.

---

This is yet another instance where we have to consider the fact that the same event handler code runs for both app and coordinator even though certain parts need to be specific to each role. I've increased the implicit annoyance counter of https://github.com/get10101/10101/issues/885.

holzeis commented on crates/ln-dlc-node/src/ln/event_handler.rs:
```diff
@@ -458,8 +458,20 @@ where
+                        {
+                            // If this fails it's either because the intercepted HTLC was already
+                            // failed or already claimed
+                            let _ = self.channel_manager.fail_intercepted_htlc(*intercept_id);
```
🤔 Shouldn't we at least log the error if we get one?

luckysori replied:
I would prefer not to. There is nothing we can do with this information and the HTLC will have already been handled properly anyway.

holzeis reviewed (approved):
Very nice! It would be great if we could add or extend a test for that.

e.g. something similar like that `open_jit_channel_with_disconnected_payee()` with a disconnect after the `Event::FundingGenerationReady`. We could provide an external eventhandler in the tests that calls the disconnect upon that event.

klochowicz reviewed (approved):
LGTM! 👍

luckysori commented:
> Very nice! It would be great if we could add or extend a test for that.
> 
> e.g. something similar like that `open_jit_channel_with_disconnected_payee()` with a disconnect after the `Event::FundingGenerationReady`. We could provide an external eventhandler in the tests that calls the disconnect upon that event.

I've added a couple of tests with https://github.com/get10101/10101/pull/979/commits/bb6c7972abb4f387dec3102c957a6a4b8d7c76d6. I would normally try to include these in the original patches, but the interactive rebase is a bit annoying and I don't want to squash everything in between.

holzeis commented:
> > Very nice! It would be great if we could add or extend a test for that.
> > e.g. something similar like that `open_jit_channel_with_disconnected_payee()` with a disconnect after the `Event::FundingGenerationReady`. We could provide an external eventhandler in the tests that calls the disconnect upon that event.
> 
> I've added a couple of tests with [bb6c797](https://github.com/get10101/10101/commit/bb6c7972abb4f387dec3102c957a6a4b8d7c76d6). I would normally try to include these in the original patches, but the interactive rebase is a bit annoying and I don't want to squash everything in between.

Nice, thank you. Would it be possible to add a test case where we triggered the channel creation already, but the app disconnects in between? We might be able to do that by listening to the `Event:FundingGeneratedEvent` from the tests and disconnect the app from the coordinator then. So the channel should not become ready but closed. (I guess)

luckysori commented:
> > > Very nice! It would be great if we could add or extend a test for that.
> > > e.g. something similar like that `open_jit_channel_with_disconnected_payee()` with a disconnect after the `Event::FundingGenerationReady`. We could provide an external eventhandler in the tests that calls the disconnect upon that event.
> > 
> > 
> > I've added a couple of tests with [bb6c797](https://github.com/get10101/10101/commit/bb6c7972abb4f387dec3102c957a6a4b8d7c76d6). I would normally try to include these in the original patches, but the interactive rebase is a bit annoying and I don't want to squash everything in between.
> 
> Nice, thank you. Would it be possible to add a test case where we triggered the channel creation already, but the app disconnects in between? We might be able to do that by listening to the `Event:FundingGeneratedEvent` from the tests and disconnect the app from the coordinator then. So the channel should not become ready but closed. (I guess)

Added another test using the trick you described! Merging now.

luckysori commented:
bors r+

luckysori commented:
bors retry
